hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/metrics/MetricsRenderPlugin.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.impl.metrics;
+
+public interface MetricsRenderPlugin {
```

Comment:
can we extend ProbeRenderer instead (with default implementations for renderNoValue and renderException) and also add some javadoc for the methods?

Reply:
We handle the exceptions in a common way. `renderException` is supposed to send the exception text to the target (to MC or to JMX bean). In both our cases we just log the exception, if each plugin was a `ProbeRenderer` itself, the exception would be logged twice. 

`renderNoValue` is used in very special case: when the metric was deregistered while iterating the metrics. There's no point in having it.

Reply:
can we rename MetricsRendererPlugin -> MetricsConsumer and the methods appropriately? Rendering implies showing something visually and plugin implies dynamic discovery/registration..